### PR TITLE
Add comments to the runtime_build script

### PR DESCRIPTION
**Pull request by mamoodi**
**What is the problem that this fixes or functionality that this introduces? Does it fix any open issues?**

Add comments to the runtime_build script so it can be followed a little bit easier. I wanted to understand this script as I was looking into the flaky test and thought maybe this will help the next person.

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

---
**Other references**

### CODE REVIEW

File: opendevin/runtime/utils/runtime_build.py [hunk 1]
```diff
@@ -80,13 +92,23 @@ def _generate_dockerfile(
     skip_init: bool = False,
     extra_deps: str | None = None,
 ) -> str:
-    """Generate the Dockerfile content for the eventstream runtime image based on user-provided base image."""
+    """Generate the Dockerfile content for the runtime image based on the base image.
+
+    Parameters:
+    - base_image (str): The base image provided for the runtime image
+    - skip_init (boolean):
+    - extra_deps (str):
```
Comment (mamoodi) [inline opendevin/runtime/utils/runtime_build.py:100]: Didn't figure these ones out.

File: opendevin/runtime/utils/runtime_build.py [hunk 2]
```diff
@@ -101,26 +123,37 @@ def prep_docker_build_folder(
...
 ) -> str:
-    """Prepares the docker build folder by copying the source code and generating the Dockerfile.
+    """Prepares a docker build folder by copying the source code and generating the Dockerfile
 
-    Return the MD5 hash of the directory.
+    Parameters:
+    - dir_path (str): The build folder to place the source code and Dockerfile
+    - base_image (str): The base Docker image to use for the Dockerfile
+    - skip_init (str):
+    - extra_deps (str):
```
Comment (mamoodi) [inline opendevin/runtime/utils/runtime_build.py:132]: Same here

File: opendevin/runtime/utils/runtime_build.py [hunk 3]
```diff
@@ -241,12 +286,22 @@ def build_runtime_image(
...
     force_rebuild: bool = False,
 ) -> str:
-    """Build the runtime image for the OpenDevin runtime.
+    """Prepares the final docker build folder.
+    If dry_run is False, it will also build the OpenDevin runtime Docker image using the docker build folder.
+
+    Parameters:
+    - base_image (str): The name of the base Docker image to use
+    - docker_client (docker.DockerClient): The Docker client
+    - extra_deps (str):
```
Comment (mamoodi) [inline opendevin/runtime/utils/runtime_build.py:295]: This one too

**Review by neubig — APPROVED** [review]
Looks great, thanks! I think this is a net positive, and we can fill the other ones in in the future.

**Pull request merged**